client/tests/test_zksnark.py

```diff
@@ -2,18 +2,18 @@
 #
 # SPDX-License-Identifier: LGPL-3.0+
 
-from zeth.core import zksnark
-from zeth.core.zksnark import Groth16SnarkProvider
+from zeth.core.zksnark import IZKSnarkProvider, GenericG1Point, GenericG2Point, \
+    IVerificationKey, ExtendedProof, Groth16, group_point_g1_to_proto, \
+    group_point_g1_from_proto, group_point_g2_to_proto, group_point_g2_from_proto
 from zeth.api import ec_group_messages_pb2
-import json
 from unittest import TestCase
 
 
 # pylint: disable=line-too-long
-VERIFICATION_KEY_BLS12_377_GROTH16 = {
+VERIFICATION_KEY_BLS12_377_GROTH16 = Groth16.VerificationKey.from_json_dict({
     "alpha": [
         "0x009d7309d79d5215384a7a9a1d9372af909582781f388a51cb833c87b8024519cf5b343cb35d49a5aa52940f14b7b8e7",  # noqa
-        "0x012816ef6069ef1e40eaab0a111f9b98b276dbf2a3209d788eb8ce635ce92a29c2bcdaa3bb9b375a8d3ee4325c07f4ea"  # noqa
+        "0x012816ef6069ef1e40eaab0a111f9b98b276dbf2a3209d788eb8ce635ce92a29c2bcdaa3bb9b375a8d3ee4325c07f4ea",  # noqa
```

Comment:
Why did you need to add this comma here?

Reply:
It's not needed as such.  Many of our lists and dictionaries have a trailing comma, so I added it to match those.

Reply:
Adding this comma here contrasts with the rest of the block actually: https://github.com/clearmatics/zeth/blob/client-proof-types/client/tests/test_zksnark.py#L13-L39
which itself contrasts with: https://github.com/clearmatics/zeth/blob/client-proof-types/client/tests/test_zksnark.py#L172-L198 (where you add final commas everywhere)

Reply:
updated.  (yes, there are examples of each - some data was copy-pasted from json)